### PR TITLE
Remove the logging system in favor of the Log crate

### LINKED ISSUE #71: Remove the logging system in favor of the Log crate
[Log](https://crates.io/crates/log) has basically the same purpose as the loging system in the server, namely a common, user defined logger. I see no reason to keep the logging system around when Log is widely adopted and has an even more convenient API.

What should be done is:

 * Remove the `log` module.
 * Remove the presence of the logger from the server, every context and every response type.
 * Include Log as a dev-dependency and use it in examples.

The reason for using it in the examples is that I don't want to promote `println!` logging. Forced log printing is nasty and we don't want to look nasty, do we?

It's not the smallest task, but it shouldn't be too complicated.

### PR DESCRIPTION
**Pull request by kvikas**
1. Removed `log` module.
2. Added `Log` and `env_logger` as dev dependency.
3. s/println!/log macros/ in example servers.
4. Removed logging section from doc

Fixes #71

### CODE REVIEW

**Comment by Ogeon** [comment]
This looks good, so far. Replacing the `println!` lines in the examples with `info!` was a nice idea. Will they still print by default (I haven't yet tested, myself)? This looks much better than what we had before.

**Comment by kvikas** [comment]
If no environment is set it defaults to logging only error messages. Since the messages are logged with `info!()`, user has to run `RUST_LOG=info ./target/debug/examples/tiny_hello` to print all messages. Going by source of `env_logger` doesn't look like there is a way to override this behavior.

**Comment by Ogeon** [comment]
Hmm... Ok, those messages are supposed to be visible by default, so I think it's best to `println!` them, after all.

**Comment by kvikas** [comment]
Done and done.

**Comment by Ogeon** [comment]
Nice! It looks like that may be it. There are only a couple of formalities left:

I would like you to edit your PR description here to not have the "TODO" part, and change "Issue" to "Closes" or "Fixes", as I did for you in #73. The description goes into the final merge description, so it's nice if it's descriptive and doesn't contain uncompleted todo lists.

I would also appreciate it if you could rebase your commits so that the fixes are combined with the other two, or maybe just squash them together into one commit. It's easier to follow the changes if there are as few fixup commits as possible.

I will take another look at this tomorrow, when I'm not as tired, and see if I have missed something before I merge it. Thank you for doing this, it's really helpful.

File: Cargo.toml [hunk 1]
```diff
@@ -49,6 +49,8 @@ optional = true
 [dev-dependencies]
 unicase = "1.0"
 tempdir = "0.3"
```
Comment (Ogeon) [inline Cargo.toml]: We don't need `tempdir` any more. It was only used in the tests for `log`.

File: examples/filters.rs [hunk 2]
```diff
@@ -112,9 +118,9 @@ impl RequestLogger {
 
 impl ContextFilter for RequestLogger {
     ///Count requests and log the path.
-    fn modify(&self, ctx: FilterContext, context: &mut Context) -> ContextAction {
+    fn modify(&self, _ctx: FilterContext, context: &mut Context) -> ContextAction {
         *self.counter.write().unwrap() += 1;
-        ctx.log.note(&format!("Request #{} is to '{}'", *self.counter.read().unwrap(), context.uri));
+        info!("Request #{} is to '{}'", *self.counter.read().unwrap(), context.uri);
```
Comment (Ogeon) [inline examples/filters.rs]: `debug!` may actually be more suitable here

Comment (Ogeon) [inline examples/filters.rs]: ...or maybe not. That won't print by default, either. This is such an important part of the example. Maybe we should add a hit to the welcome message that they can run it with debug prints. What do you think?

Comment (kvikas) [inline examples/filters.rs]: Yeah I think that is good idea. We can actually conditionally print. If program is run with a log level that will not print this line, we can print instructions to run with log level debug to see some information.

Comment (Ogeon) [inline examples/filters.rs]: A simple line that says something like "Run this example with the environment variable 'RUST_LOG' set to 'debug' to see the debug prints." would probably be enough.

**Comment by Ogeon** [comment]
I just realized how much Hyper prints in debug mode... That example is a little special, anyway, so I think this is good for now. Thank you!

@homu r+

**Pull request merged**